### PR TITLE
Add missing operator and operand RBAC

### PR DESCRIPTION
**Pull request by Danil-Grigorev**

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
/approve

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ spec:
       serviceAccountName: cluster-cloud-controller-manager
       containers:
       - name: cluster-cloud-controller-manager
-        image: registry.svc.ci.openshift.org/openshift:cluster-cloud-controller-manager
+        image: registry.svc.ci.openshift.org/openshift:cluster-cloud-controller-manager-operator
```
Comment (Fedosin) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:25]: Shouldn't the image come from quay.io?

File: manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml [hunk 2]
```diff
@@ -115,10 +114,26 @@ metadata:
...
 roleRef:
-  apiGroup: rbac.authorization.k8s.io
   kind: Role
   name: cluster-cloud-controller-manager
 subjects:
   - kind: ServiceAccount
     namespace: openshift-cloud-controller-manager-operator
     name: cluster-cloud-controller-manager
+
+# ---
```
Comment (Fedosin) [inline manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml]: Do we need to keep the commented RoleBinding here?

File: manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml [hunk 3]
```diff
@@ -121,19 +121,19 @@ subjects:
...
-# roleRef:
-#   kind: Role
-#   name: cluster-cloud-controller-manager
-# subjects:
-#   - kind: ServiceAccount
-#     namespace: openshift-cloud-controller-manager-operator
-#     name: cluster-cloud-controller-manager
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
```
Comment (Fedosin) [inline manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml:126]: Probably ClusterRoleBinding...

File: manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml [hunk 4]
```diff
@@ -121,19 +121,19 @@ subjects:
...
+metadata:
+  name: cluster-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+roleRef:
+  kind: ClusterRole
+  name: admin
```
Comment (Fedosin) [inline manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml:135]: Where is this cluster role defined?

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml (hunk 1 above)
Comment (Danil-Grigorev) [reply manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:25]: No, this tag should be substituted with real image hash coming image-registry during image injection. The image is build for us by CI, and the operator is not intended to run outside openshift cluster.

File: manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml (hunk 2 above)
Comment (Danil-Grigorev) [reply manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml]: Accidentally slipped between commits, will squash it in a second.

File: manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml (hunk 4 above)
Comment (Danil-Grigorev) [reply manifests/0000_26_cloud-controller-manager-operator_02_rbac.yaml:135]: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles One of the default options.

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**